llvm/lib/Target/AIE/AIELegalizerHelper.cpp

```diff
@@ -182,9 +194,30 @@ bool AIELegalizerHelper::legalizeG_BUILD_VECTOR(LegalizerHelper &Helper,
   //    https://github.com/llvm/llvm-project/pull/84538
   if (DstVecSize == 256) {
     const Register UnusedSubReg = MRI.createGenericVirtualRegister(DstVecTy);
-    MIRBuilder.buildUnmerge({DstReg, UnusedSubReg}, Src);
+    MIRBuilder.buildUnmerge({UnusedSubReg, DstReg}, Src);
```

Comment:
Why did this change?

Reply:
Because now we need higher 256 bits (As we are using VPUSH.hi now)

Reply:
Maybe add that as a comment? We are using `getGenericAddVectorEltOpcode()` which adds elements from the hi bits, so the 256-bit result will be in the hi part of the 512-bit vector.